mesonbuild/backend/ninjabackend.py

```diff
@@ -1103,6 +1103,22 @@ def generate_vala_compile(self, target, outfile):
                 # Install GIR to default location if requested by user
                 if len(target.install_dir) > 3 and target.install_dir[3] is True:
                     target.install_dir[3] = os.path.join(self.environment.get_datadir(), 'gir-1.0')
+            # Generate header for internal symbols if requested by user
+            if isinstance(target.vala_internal_header, str):
+                internalheadername = os.path.join(self.get_target_dir(target), target.vala_internal_header)
+                args += ['--internal-header', internalheadername]
+                valac_outputs.append(internalheadername)
+                target.outputs.append(target.vala_internal_header)
+                if len(target.install_dir) > 4 and target.install_dir[4] is True:
+                    target.install_dir[4] = self.environment.get_includedir()
```

Comment:
Generation of these internal headers/definitions looks good, but these should not be installed because they are not for public use (by definition). If you allow people outside your own project to use your private structures, you can't change them ever again.